### DISCUSSION TITLE
How to specify source of BTF？

### DISCUSSION THREAD

**Discussion by cigarl** [cilium/ebpf#1467] [Q&A] [answered]
I found the previous [commit record](https://github.com/cilium/ebpf/pull/287), but I didn't find it in the latest code.
And i find [btf/vmlinx](https://github.com/cilium/ebpf/blob/8079b37aa138b38707461cb8d548edf3359d6bc2/btf/kernel.go#L98) and [other path](https://github.com/cilium/ebpf/blob/8079b37aa138b38707461cb8d548edf3359d6bc2/btf/kernel.go#L132) in latest code.

It seems that custom specifying _BTF_ is no longer allowed,or did I miss something?

**Comment by brycekahle** [comment, marked as answer]
See `ProgramOptions.KernelTypes` and `CollectionOptions.Programs.KernelTypes`https://pkg.go.dev/github.com/cilium/ebpf#ProgramOptions

**Reply by cigarl** [reply]
thank you soooo much.

BTW, It would be greater if each interface had an example of how to use the relevant parameters (_It took me a little bit to understand how to generate_ `Spec`).
